src/github.ts

```diff
@@ -1333,6 +1291,21 @@ export class GitHub {
         pullRequest.labels
       );
 
+      // attempt to enable auto-merge
+      if (options?.autoMerge) {
```

Comment:
I think we only wanted to auto-merge if the conventional commit messages matched certain patterns, namely:

```
fix:
fix(*):
feat(api):
```
(Personally I think we should auto-merge on _any_ `feat` PR, but `feat(api)` is a compromise I think @RobertCraigie  and I have agreed on; I feel much more strongly about `fix`).

We **don't** want to auto-merge if it's a major version bump.

We also **do** want to auto-merge if the release PR has been open a while (IMO, >72h is OK).

Does this sound relatively straightforward to implement?

Reply:
You will see during your review, I do a check based on the version bump. What I implement is "only use auto-merge for minor, patch, or build". Which should catch `fix` (that's a "patch") and `feat` (that's a "minor"). Would you prefer explicit checks on commit messages?

Reply:
> We also do want to auto-merge if the release PR has been open a while (IMO, >72h is OK).

I feel that a scheduled (maybe daily?) `gh pr merge` would make more sense for that type of thing, release-please wouldn't really be the good place for that type of check IMHO.

Reply:
Just once per day pull all open PR with label `autorelease: pending` for all customers, and call `gh pr merge` on them.

Reply:
> Which should catch fix (that's a "patch") and feat (that's a "minor"). Would you prefer explicit checks on commit messages?

This is fine with me; I think Robert had wanted to avoid shipping "less important" feat's and stuff like internal chores, which tend to result in patch releases but can be noisy. I think it's fine for that filtering to happen in a follow-on ticket/PR though! Would you mind filing it?

I also think that you're correct that the scheduled release should happen in a cron instead of the "if a push comes in after N hours" algorithm I proposed; that can also be a follow-on ticket (please file!).

Reply:
Yeah I'm okay with `feat(api):` commits being auto-released.

I'd honestly much rather prefer tagging commits explicitly or just a button somewhere that merges all feasible release PRs for other cases though. I feel less strongly about including `fix` by default, I just think the end UX would not be good if we're spamming releases.

I do think that the filtering should happen in this PR though.

Reply:
Okay, I'm cool with blocking this PR on filtering commit messages more granularly, per the above algorithm!

Reply:
Ticket for the cron-job for stalled release PR: https://linear.app/stainless/issue/STA-2330/periodically-merge-staled-release-prs